src/gc/config.d

```diff
@@ -40,6 +42,8 @@ struct Config
     {
         import core.internal.traits : externDFunc;
 
+        __gshared linkDummy = &_dummy_ref_to_link_gc_register_constructor;
```

Comment:
Wouldn't it be simpler to just call `_d_register_conservative_gc()` and `_d_register_manual_gc()` explicitely. You can still  override these symbols in the executable if you don't want to register them.

Reply:
The C object file wouldn't be referenced and doesn't get linked into the application.

Reply:
What C object file? Just call _d_register_conservative_gc() and _d_register_manual_gc() right here.

Reply:
Just noticed yesterday that you can also reimplement gc_init in the executable registering the desired or new GCs.

Reply:
That wouldn't allow configuring by application users. Think of how Java GC's are often selected/optimized according to specific loads.
Wouldn't work with druntime as DLL or with hidden symbols in shared libraries.
Also many programmers hardly have an idea how linkers work.

Reply:
Agreed, replacing gc_init is a bit blunt and doesn't allow adding to existing GCs. It's currently the only way to remove a GC from the binary.

The current of `_dummy_ref_to_link_gc_register_constructor` seems gratuitious, though. Just registering the GCs here should be fine.